src/treeview_help.ml

```diff
@@ -0,0 +1,118 @@
+        ]
+      ()
+  in
+  Vscode.TreeItem.set_iconPath item icon;
+  Vscode.TreeItem.set_command item command;
+  item
+
+let discord_item ~extension_path =
+  let icon =
+    `LightDark
+      Vscode.TreeItem.LightDarkIcon.
+        { light = `String (extension_path ^ "/assets/discord-light.svg")
+        ; dark = `String (extension_path ^ "/assets/discord-dark.svg")
+        }
+  in
+  let label = Vscode.TreeItemLabel.create ~label:"Chat on Discord" () in
+  let item = Vscode.TreeItem.make ~label () in
+  let command =
+    Vscode.Command.create ~title:"Open" ~command:"vscode.open"
+      ~arguments:
+        [ Vscode.Uri.parse "https://discord.gg/cCYQbqN" () |> Vscode.Uri.t_to_js
+        ]
+      ()
+  in
+  Vscode.TreeItem.set_iconPath item icon;
+  Vscode.TreeItem.set_command item command;
+  item
+
+let discuss_item ~extension_path =
+  let icon =
+    `LightDark
+      Vscode.TreeItem.LightDarkIcon.
+        { light = `String (extension_path ^ "/assets/chat-light.svg")
+        ; dark = `String (extension_path ^ "/assets/chat-dark.svg")
+        }
+  in
+  let label =
+    Vscode.TreeItemLabel.create ~label:"Ask a question on Discuss" ()
+  in
+  let item = Vscode.TreeItem.make ~label () in
+  let command =
+    Vscode.Command.create ~title:"Open" ~command:"vscode.open"
+      ~arguments:
+        [ Vscode.Uri.parse "https://discuss.ocaml.org/" () |> Vscode.Uri.t_to_js
+        ]
+      ()
+  in
+  Vscode.TreeItem.set_iconPath item icon;
+  Vscode.TreeItem.set_command item command;
+  item
+
+let github_item ~extension_path =
+  let icon =
+    `LightDark
+      Vscode.TreeItem.LightDarkIcon.
+        { light = `String (extension_path ^ "/assets/github-light.svg")
+        ; dark = `String (extension_path ^ "/assets/github-dark.svg")
+        }
+  in
+  let label = Vscode.TreeItemLabel.create ~label:"Open an issue on Github" () in
+  let item = Vscode.TreeItem.make ~label () in
+  let command =
+    Vscode.Command.create ~title:"Open" ~command:"vscode.open"
+      ~arguments:
+        [ Vscode.Uri.parse "https://github.com/ocamllabs/vscode-ocaml-platform"
+            ()
+          |> Vscode.Uri.t_to_js
+        ]
+      ()
+  in
+  Vscode.TreeItem.set_iconPath item icon;
+  Vscode.TreeItem.set_command item command;
+  item
+
+let items ~extension_path =
+  [ tutorial_item ~extension_path
+  ; discord_item ~extension_path
+  ; discuss_item ~extension_path
+  ; github_item ~extension_path
+  ]
+
+let getTreeItem ~extension_path:_ ~element = Promise.return element
+
+let getChildren ~extension_path ~element =
+  match element with
+  | None -> `Promise (Promise.return (Some (items ~extension_path)))
+  | Some _ -> `Promise (Promise.return (Some []))
+
+let register extension =
+  let extension_path = Vscode.ExtensionContext.extensionPath extension in
+  let treeDataProvider =
+    Vscode.TreeDataProvider.create
+      ~getTreeItem:(getTreeItem ~extension_path)
+      ~getChildren:(getChildren ~extension_path)
+      ()
+  in
+  let _ =
```

Comment:
Is it correct to drop the disposable here?

Reply:
We have no use for it, I don't think this has any impact.

Reply:
To restart the extension, we need to dispose of it and all the resources it acquired. So I think we should also dispose the tree acquired here as well. See `Extension_instance.disposable`.

Reply:
In previous versions of the plugin, we'd drop these disposables and it would make restarting the plugin after an upgrade wonky.

Reply:
Done, thanks :)